infor-design/enterprise
# 4590/4952 - App Menu trigger button/filtering improvements

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes issues related to the App Menu Trigger button on Module Tabs, as well as filtering issues in the App Menu itself:

- Adds a setting to the Tabs component to allow Module Tabs instances with an App Menu trigger enabled to optionally hide the text present inside the trigger button. (#4590)
- Fixes a regression in the App Menu filtering logic that was causing accordion items that matched the filter to incorrectly be visually hidden. (#4592)
- Changed the App Menu filter logic to hide all items when filtering on a term that matches no items.  Previously, if attempting to filter on a term that matched no items, all items would be displayed. (#4592)

**Related github/jira issue (required)**:
Fixes #4590
Fixes #4592
Related to #4052

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the demoapp, then:

_Test Module Tabs App Menu Trigger:_
- Open http://localhost:4000/components/tabs-module/example-app-menu-button-audible.html?layout=nofrills
- Observe that the app menu trigger button has no text visible.
- Open a screen reading tool such as Apple Voiceover and navigate to the App Menu trigger.  It should read the word "Menu" in some way, which is still present in the DOM for accessibility purposes.

_Test App Menu Filtering:_
- Open http://localhost:4000/components/applicationmenu/example-filterable?theme=uplift&variant=light
- In the app menu searchfield, type "#3".  After filtering completes, the results should include: "Role #1 Task #3", "Role #2 Task #3", and "Role #3".
- Clear the searchfield and type "zzzzz".  After filtering completes, no items should be present in the filter results.

**Included in this Pull Request**:

@whernebrink @Milan1983
